### PR TITLE
refactor cohort-grid

### PR DESCRIPTION
**Pull request by beaugogh**
also containing the fix for the ticket "Go to Setting and back to workspace will duplicate workspace"

### CODE REVIEW

File: src/app/components/cohort-charts/dc-charts.service.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ angular.module('transmartBaseUi').factory('DcChartsService', [function () {
         var width = (_chart.gridInfo.sizeX * _chart.gridInfo.curColWidth) - 50;
         var height = (_chart.gridInfo.sizeY * _chart.gridInfo.curRowHeight) - 60;
 
-        if(width > 0 && height > 0) {
+        if(width !== Infinity & height !== Infinity & width > 0 && height > 0) {
```
Comment (rnugraha) [inline src/app/components/cohort-charts/dc-charts.service.js]: can we have proper bracketing here and reason why using bitwise and logical AND

File: src/app/components/cohort-grid/cohort-grid.service.js [hunk 2]
```diff
@@ -41,36 +41,40 @@ angular.module('transmartBaseUi').factory('CohortGridService', ['$timeout', func
...
+        columnDefs.push({field: 'cohort-panel', width: service.WIDTH_PER_COLUMN});
         return columnDefs;
     };
 
     /**
      * Format data and labels for the gridview's data
      * @memberof CohortGridService
      * @param subjects
-     * @param headers
+     * @param labels
```
Comment (rnugraha) [inline src/app/components/cohort-grid/cohort-grid.service.js]: looks to me the param is now only one, if this should be or then probably should be 

@param subjects or labels

File: src/app/components/cohort-selection/cohort-selection.tpl.html [hunk 3]
```diff
@@ -55,6 +55,10 @@
                 popupDelay="1000"><span class="glyphicon glyphicon-remove"></span></button>
     </div>
 
+    <div>
+        <h5><span class="cohort-name-label pull-right"> Cohort-{{+index+1}} </span></h5>
```
Comment (rnugraha) [inline src/app/components/cohort-selection/cohort-selection.tpl.html:59]: `{{+index+1}}`

this is abit worrying. What is this and why?

Comment (beaugogh) [reply src/app/components/cohort-selection/cohort-selection.tpl.html:59]: to ensure that index is used as a number not a string or sth. else, index is an attribute in the cohort-selection directive

File: src/app/components/cohort-grid/cohort-grid.service.js (hunk 2 above)
Comment (beaugogh) [reply src/app/components/cohort-grid/cohort-grid.service.js]: yes, the param should be removed

File: src/app/components/cohort-charts/dc-charts.service.js (hunk 1 above)
Comment (beaugogh) [inline src/app/components/cohort-charts/dc-charts.service.js]: will change this

Comment (rnugraha) [reply src/app/components/cohort-charts/dc-charts.service.js]: @beaugogh is this changed?

**Review by beaugogh — COMMENTED** [review]
test

File: src/app/components/cohort-charts/dc-charts.service.js [hunk 4]
```diff
@@ -39,7 +40,11 @@ angular.module('transmartBaseUi').factory('DcChartsService', [function () {
         var width = (_chart.gridInfo.sizeX * _chart.gridInfo.curColWidth) - 50;
         var height = (_chart.gridInfo.sizeY * _chart.gridInfo.curRowHeight) - 60;
 
-        if(width !== Infinity & height !== Infinity & width > 0 && height > 0) {
+        if(width > 0 &&
```
Comment (rnugraha) [inline src/app/components/cohort-charts/dc-charts.service.js]: ok

**Pull request merged**